.github/scripts/compare_build_time.sh

```diff
@@ -0,0 +1,106 @@
+cargo bench --bench request_template_bench_iai-callgrind -- --save-baseline change >> benches/iai-callgrind/new_benchmarks.txt
+sed -i 's/ \{1,\}\([0-9]\)/\1/g' benches/iai-callgrind/new_benchmarks.txt
+file2="benches/iai-callgrind/new_benchmarks.txt"
+
+config_file="benches/iai-callgrind/benchmarks.cfg" # to add more benchmarks add in this file
+# Read benchmarks from the configuration file
+readarray -t benchmarks < "$config_file"
+
+attributes=("Instructions" "L1 Hits" "L2 Hits" "RAM Hits")
+
+fail_ci=0
+
+for bench in "${benchmarks[@]}"; do
+    x=0
+    y=0
+
+    # Check attribute changes
+    for attribute in "${attributes[@]}"; do
+        value1=$(grep -A5 "$bench" "$file1" | grep -Po "${attribute}:\K\d+")
+        value2=$(grep -A5 "$bench" "$file2" | grep -Po "${attribute}:\K\d+")
+
+        if [ -n "$value1" ] && [ -n "$value2" ]; then
+            if [ "$value1" -ne 0 ]; then
+                percent_change=$(( ((value2 - value1) * 100) / value1 ))
+            else
+                percent_change="nan"
+            fi
+
+            if ((percent_change > 10)); then
+                echo "$bench $attribute has a change of $percent_change%, failing CI."
+                fail_ci=1
+            else
+                echo "$bench $attribute has a change of $percent_change%, within CI limits."
+            fi
+        else
+            echo "Values not found for $bench $attribute"
+        fi
+    done
+
+    # Check performance metric changes 
+    # Total read+write = L1 Hits + L2 Hits + RAM Hits.
+    # Estimated Cycles = L1 Hits + 5 × (L2 Hits) + 35 × (RAM Hits).
+    for file in "$file1" "$file2"; do
+        l1_hits=$(grep -A5 "$bench" "$file" | grep -Po "L1 Hits:\K\d+")
+        l2_hits=$(grep -A5 "$bench" "$file" | grep -Po "L2 Hits:\K\d+")
+        ram_hits=$(grep -A5 "$bench" "$file" | grep -Po "RAM Hits:\K\d+")
+
+        if [ "$x" -ne 0 ]; then
+            total_read_write=$((l1_hits + l2_hits + ram_hits))
+            estimated_cycles=$((l1_hits + 5 * l2_hits + 35 * ram_hits))
+
+            p1=$(( ((total_read_write - x) * 100) / x ))
+            echo "$bench Total read+write has a change of $p1%"
+
+            if ((p1 > 10)); then
+                echo "$bench Total read+write has a change greater than 10%, failing CI."
+                fail_ci=1
+            fi
+
+            p2=$(( ((estimated_cycles - y) * 100) / y ))
+            echo "$bench Estimated Cycles has a change of $p2%"
+
+            if ((p2 > 10)); then
+                echo "$bench Estimated Cycles has a change greater than 10%, failing CI."
+                fail_ci=1
+            fi
+        else
+            total_read_write=$((l1_hits + l2_hits + ram_hits))
+            estimated_cycles=$((l1_hits + 5 * l2_hits + 35 * ram_hits))
+        fi
+
+        x=$((x + total_read_write))
+        y=$((y + estimated_cycles))
+    done
```

Comment:
for previous pr @tusharmath has suggested to use `cargo-script` for such kind of scripts.

wdyt about it? simple bash scripts are ok overall but when they get more complex it's really a burden to support it. Or consider another script language that is more reader friendly

Reply:
i have shorten the code and it looks simple to me  
this code will always work in comparing iai-callgrind and will never need any changes

Reply:
well, the experience says that any code may require changes despite initial intentions for it

Reply:
ok, but does the current code looks simple to deal with or should use cargo script